go-selinux/selinux_linux.go

```diff
@@ -445,6 +460,186 @@ func ComputeCreateContext(source string, target string, class string) (string, e
 	return readWriteCon(filepath.Join(getSelinuxMountPoint(), "create"), fmt.Sprintf("%s %s %d", source, target, classidx))
 }
 
+/*
+catsToBitset stores categories in a bitset.
+*/
+func catsToBitset(cats string) (*bitset.BitSet, error) {
+	bitset := &bitset.BitSet{}
+
+	catlist := strings.Split(cats, ",")
+	for _, r := range catlist {
+		ranges := strings.Split(r, ".")
+		if len(ranges) > 1 {
+			catstart, err := strconv.Atoi(string([]rune(ranges[0])[1:]))
+			if err != nil {
+				return nil, err
```

Comment:
Should some of these be errors.Wrap?  What do the errors look like when you give bogus data?

Reply:
Should we check the catstart and catend to see if they are within the valid range of categories supported by the library? (1024)

Reply:
> Should some of these be errors.Wrap? What do the errors look like when you give bogus data?

Thanks for the review.

Any objection to using `fmt.Errorf()` and `errors.Is()` for wrapping? 



>Should we check the catstart and catend to see if they are within the valid range of categories supported by the library? (1024)

We have use cases where we need more than 1024 categories. Would you prefer a maxCats  argument to this function or to parse this from `/sys/fs/selinux/initial_contexts/kernel`?

Reply:
I would prefer to continue using errors.Wrapf, since the code already uses it.

Reply:
I think eventually we should switch to `fmt.Errorf` and `errors.Is/.As` since it's now a standard which does not require a third-party package. Having said that, such a change needs to be performed at once and repo-wise.